akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -342,6 +345,15 @@ object ShardRegion {
     var remaining = entities
 
     def receive = {
+      case ReceiveTimeout ⇒
+        log.warning("HandOffStopMessage[{}] is not handled by some of the entites of the `{}` shard, " +
+          "force stopping the remaining entities.", stopMessage.getClass.getName, shard)
+
+        remaining.foreach {
+          ref ⇒
+            context stop ref
```

Comment:
So this will forcefully stop the actor, trigger a `Terminated` because we're watching it, and make sure `ShardStopped` is triggered when all actors are terminated? Seems OK I guess.

Reply:
Yes.

Reply:
right, the important part is that we don't act until we got all `Terminated`